airflow/providers/databricks/sensors/databricks.py

```diff
@@ -0,0 +1,137 @@
+        self.client_parameters = client_parameters or {}
+
+    def _get_hook(self) -> DatabricksSqlHook:
+        return DatabricksSqlHook(
+            self.databricks_conn_id,
+            http_path=self._http_path,
+            session_configuration=self.session_config,
+            sql_endpoint_name=self._sql_endpoint_name,
+            http_headers=self.http_headers,
+            catalog=self.catalog,
+            schema=self.schema,
+            **self.client_parameters,
+        )
+
+
+class DatabricksPartitionTableSensor(DatabricksBaseSensor):
+    """
+    Waits for a partition to show up in Databricks.
+
+    :param table_name (str): The name of the table to wait for.
+    :param partition_name (str): The partition clause to wait for.
+    :param database_name (str): The name of the database in Databricks. It uses 'default' if nothing is provided
+    :param databricks_conn_id (str): Reference to the :ref:`Databricks connection <howto/connection:databricks>`.
+    """
+    template_fields: Sequence[str] = (
+        'database_name',
+        'table_name',
+        'partition_name',
+    )
+
+    def __init__(self, *,
+                 databricks_conn_id: str,
+                 table_name: str,
+                 partition_name: str,
+                 database_name: Optional[str] = 'default',
+                 **kwargs: Any):
+        super().__init__(**kwargs)
+        self.databricks_conn_id = databricks_conn_id
+        self.table_name = table_name
+        self.partition_name = partition_name
+        self.database_name = database_name
+
+    def poke(self, context: Context) -> bool:
+        hook = self._get_hook()
+        _, result = hook.run(f'SHOW PARTITIONS {self.database_name}.{self.table_name}')
+        record = result[0] if result else {}
+        return self.partition_name in record
+
+
+class DatabricksDeltaTableChangeSensor(DatabricksBaseSensor):
+    """
+    Waits for Delta table event
+
+    :param table_name (str): The name of the table to wait for, supports the dot
+    :param timestamp (datetime): The timestamp that will be used to filter new events.
+    :param database (Optional[str]): The name of the database in Databrick. It uses 'default' if nothing is provided
+    :param databricks_conn_id (str): Reference to the :ref:`Databricks connection <howto/connection:databricks>`.
+    """
+    template_fields: Sequence[str] = (
+        'database_name',
+        'table_name',
+    )
+
+    def __init__(self, *,
+                 databricks_conn_id: str,
+                 table_name: str,
+                 timestamp: datetime,
+                 database_name: Optional[str] = 'default',
+                 **kwargs: Any):
+        super().__init__(**kwargs)
+        self.databricks_conn_id = databricks_conn_id
+        self.table_name = table_name
+        self.timestamp = timestamp
+        self.database_name = 'default' if not database_name else database_name
+
+    def poke(self, context: Context) -> bool:
```

Comment:
I would generalize this into the `DatabricksSqlSensor` and just pass different conditions